CMakeLists.txt

```diff
@@ -0,0 +1,112 @@
+cmake_minimum_required(VERSION 3.11)
+
+set(PRODUCT_NAME cretro)
+string(TOUPPER ${PRODUCT_NAME} MSG_PREFIX)
+
+project(
+    ${PRODUCT_NAME}
+    LANGUAGES C
+)
+
+set(CMAKE_C_COMPILER gcc)
+
+set(INSTALL_PATH ~/.local/bin/${PRODUCT_NAME})
+
+add_compile_options(
+    -Wall
+    -Wextra
+    -Werror
+    -Wunused
+    -Wshadow
+    -Wcast-align
+    -Wconversion
+    -Wlogical-op
+    -Wdouble-promotion
+    -fstack-protector
+    -Wstack-protector
+    -Wredundant-decls
+    -Wpacked
+    -Wmissing-declarations
+    -Wmissing-prototypes
+    -Wstrict-prototypes
+    -Wno-aggressive-loop-optimizations
+    -Wjump-misses-init
+    -Wwrite-strings
+    -Wcast-qual
+    -Wundef
+    -Wunsafe-loop-optimizations
+    -Wfloat-equal
+    -Wtrampolines
+    -Warray-bounds=2
+    -Wsuggest-attribute=cold
+    -Wsuggest-attribute=malloc
+    -Wsuggest-attribute=pure
+    -Wsuggest-attribute=const
+    -Wsuggest-attribute=noreturn
+    -Wsuggest-attribute=format
+    -Wstrict-overflow=4
+    -Wswitch-enum
+    -Wswitch-default
+    -Wmissing-include-dirs
+    -Wformat=2
+)
+
+if (CMAKE_BUILD_TYPE STREQUAL "Debug" OR CMAKE_BUILD_TYPE STREQUAL "DEBUG" OR CMAKE_BUILD_TYPE STREQUAL "debug")
+    message("[${MSG_PREFIX}] Using -g")
+    add_compile_options(-g)
+else()
+    message("[${MSG_PREFIX}] Using -O3")
+    add_compile_options(-O3)
+endif()
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
+    message("[${MSG_PREFIX}] Using clang")
+    add_compile_options(
+        -Weverything
+        -Wno-unknown-warning-option
```

Comment:
```suggestion
```
I think that we should remove this. This is preventing clang from reporting some errors. See https://github.com/Benzammour/yobemag/pull/1 where I addressed this by moving some warnings around